student-test/assignment1/Test_validate_username.c

```diff
@@ -5,18 +5,14 @@
 #include "../../assignment-autotest/test/assignment1/username-from-conf-file.h"
 
 /**
-* This function should:
-*   1) Call the my_username() function in autotest-validate.c to get your hard coded username.
-*   2) Obtain the value returned from function malloc_username_from_conf_file() in username-from-conf-file.h within
-*       the assignment autotest submodule at assignment-autotest/test/assignment1/
-*   3) Use unity assertion TEST_ASSERT_EQUAL_STRING_MESSAGE to verify the two strings are equal.  See
-*       the [unity assertion reference](https://github.com/ThrowTheSwitch/Unity/blob/master/docs/UnityAssertionsReference.md)
-*/
+ * This function should:
+ *   1) Call the my_username() function in autotest-validate.c to get your hard coded username.
+ *   2) Obtain the value returned from function malloc_username_from_conf_file() in username-from-conf-file.h within
+ *       the assignment autotest submodule at assignment-autotest/test/assignment1/
+ *   3) Use unity assertion TEST_ASSERT_EQUAL_STRING_MESSAGE to verify the two strings are equal.  See
+ *       the [unity assertion reference](https://github.com/ThrowTheSwitch/Unity/blob/master/docs/UnityAssertionsReference.md)
+ */
 void test_validate_my_username()
 {
-    /**
-     * TODO: Replace the line below with your code here as described above to verify your /conf/username.txt 
-     * config file and my_username() functions are setup properly
-     */
-    TEST_ASSERT_TRUE_MESSAGE(false,"AESD students, please fix me!");
+    TEST_ASSERT_EQUAL_STRING_MESSAGE(my_username(), malloc_username_from_conf_file(), "Username is valid");
```

Comment:
@zeog1488 what happens to the string allocated with `malloc_username_from_conf_file` here?
When will the message "Username is valid" be printed?